packages/sanes-browser-extension/src/extension/background/model/persona/index.ts

```diff
@@ -134,27 +144,51 @@ export class Persona {
 
     const profile = await UserProfile.loadFrom(db, encryptionKey);
     const signer = new MultiChainSigner(profile);
-    const managerChains = await Persona.connectToAllConfiguredChains(signer);
+    const chainStatuses: ChainStatus[] = [];
+    const managerChains = await Persona.connectToAllConfiguredChains(signer, chainStatuses);
     const manager = new SoftwareAccountManager(profile, managerChains);
 
-    return new Persona(encryptionKey, profile, signer, manager, makeAuthorizationCallbacks);
+    return new Persona(encryptionKey, profile, signer, chainStatuses, manager, makeAuthorizationCallbacks);
   }
 
   private static async connectToAllConfiguredChains(
     signer: MultiChainSigner,
+    chainStatuses: ChainStatus[],
   ): Promise<readonly SoftwareAccountManagerChainConfig[]> {
     const config = await getConfigurationFile();
 
     const out: SoftwareAccountManagerChainConfig[] = [];
     for (const chainSpec of config.chains.map(chain => chain.chainSpec)) {
       const codecType = codecTypeFromString(chainSpec.codecType);
       const connector = chainConnector(codecType, chainSpec);
-      const { connection } = await signer.addChain(connector);
-      out.push({
-        chainId: connection.chainId(),
-        algorithm: algorithmForCodec(codecType),
-        derivePath: pathBuilderForCodec(codecType),
-      });
+
+      try {
+        const { connection } = await signer.addChain(connector);
+        const chainId = connection.chainId();
+
+        out.push({
+          chainId: chainId,
+          algorithm: algorithmForCodec(codecType),
+          derivePath: pathBuilderForCodec(codecType),
+        });
+
+        chainStatuses.push({
+          name: await getChainName(chainId),
+          node: await getChainNode(chainId),
+          connected: true,
+        });
```

Comment:
This may be refactored in different ways:

- Pass both arrays as parameter
- Return an object that has both arrays
- Make 2 methods that fill the arrays and are called from `connectToAllConfiguredChains`.
- etc.

I'll think about what's more readable & maintainable.